src/formatDate.js

```diff
@@ -8,7 +8,23 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const newData = date.split(separator);
+  const newFormat = Array.from(toFormat);
+  const newSeparator = newFormat.pop();
+  const obj = {};
+
+  for (let i = 0; i < newData.length; i++) {
+    obj[fromFormat[i]] = newData[i];
+
+    if (fromFormat[i] === 'YY') {
+      obj.YYYY = newData[i] >= 30 ? '19' + newData[i] : '20' + newData[i];
```

Comment:
There is a potential issue with the year conversion logic. If `newData[i]` is less than `10`, it will not have a leading zero, which is typically present in two-digit year formats. You should ensure that the year is always two digits.